controllers/profiles/common/object_creators.go

```diff
@@ -161,13 +149,34 @@ func defaultContainer(workflow *operatorapi.SonataFlow) (*corev1.Container, erro
 	if err := mergo.Merge(defaultFlowContainer, workflow.Spec.PodTemplate.Container.ToContainer(), mergo.WithOverride); err != nil {
 		return nil, err
 	}
-	defaultFlowContainer = ConfigurePersistence(defaultFlowContainer, workflow.Spec.Persistence, defaultSchemaName, workflow.Namespace)
+	if workflow.Spec.Persistence != nil {
+		c := workflow.Spec.Persistence
+		if c.PostgreSQL == nil && plf.Spec.Persistence != nil {
+			c = &operatorapi.PersistenceOptionsSpec{}
+			if plf.Spec.Persistence.PostgreSQL != nil {
+				c.PostgreSQL = &operatorapi.PersistencePostgreSQL{
+					SecretRef: plf.Spec.Persistence.PostgreSQL.SecretRef,
+					JdbcUrl:   plf.Spec.Persistence.PostgreSQL.JdbcUrl,
```

Comment:
maybe we should add an `if len(plf.Spec.Persistence.PostgreSQL.JdbcUrl) > 0` check here first? to ensure we're not breaking the MaxItems CRD validation here?

Reply:
Updated to check the `serviceRef` for nil to fallback to the `jdbcUrl` value. Othewrise the `serviceRef` will be used and the `jdbcUrl` is not initialized.